### PR TITLE
Use a generic DOI site instead of 10.80507

### PR DESCRIPTION
**Pull request by dchiquito**
10.80507 is our test DOI site, we need to support both that and our
production site, 10.48324.

### CODE REVIEW

**Comment by satra** [comment]
@yarikoptic - yup - that's why i wanted to chat.

**Comment by yarikoptic** [comment]
What about:
- we initiate `schema-next` (so matches `schema-<version>` tags) branch off current master and push it here
- change base for #60 to be `schema-next`
- hard reset `master ` to 4f6e708f8816fb37ff88c83074715c54174857ba
- re-merge (some? all?) PRs some PRs merged since then but skip #52 which boosted version (do only local testing)
- force-push reset `master`
- re-initiate tests for this PR to just make sure we are in clear
- merge PR/release without changes to schema version

gotchas might come in changelog generation after we do merge `schema-next` into master since we are per se not interested in that merge but rather in merges/PRs into that branch.  I remember @jwodder investigated that abit awhile ago but I forgot what was the verdict.

**Comment by satra** [comment]
this PR will also require a schema version change, so before we implement a full blown refactor, we may try to cut a release to 0.5.0 with the other PR in tow before we fully implement validation against multiple schemas.

File: dandischema/models.py [hunk 1]
```diff
@@ -49,7 +49,7 @@
     "[a-f0-9]{8}[-]*[a-f0-9]{4}[-]*" "[a-f0-9]{4}[-]*[a-f0-9]{4}[-]*[a-f0-9]{12}$"
 )
 ASSET_UUID_PATTERN = r"^dandiasset:" + UUID_PATTERN
-DANDI_DOI_PATTERN = r"^10.80507/dandi\.\d{6}/\d+\.\d+\.\d+"
+DANDI_DOI_PATTERN = r"^10.[0-9]{5}/dandi\.\d{6}/\d+\.\d+\.\d+"
```
Comment (yarikoptic) [inline dandischema/models.py:52]: @satra -- do we want to limit may be to the only two we have? in the future we should relax it anyways to allow for truly arbitrary DOI prefix I guess

Comment (satra) [reply dandischema/models.py:52]: i see no reason not too at this point.

Comment (satra) [inline dandischema/models.py:52]:
```suggestion
DANDI_DOI_PATTERN = r"^10.(48324|80507)/dandi\.\d{6}/\d+\.\d+\.\d+"
```

**Pull request merged**